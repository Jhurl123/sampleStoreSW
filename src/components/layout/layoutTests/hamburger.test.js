import React from 'react';
import ReactDOM from 'react-dom'
import { BrowserRouter as Router } from 'react-router-dom';
import { mount, configure } from 'enzyme';
import Hamburger from '../menu/Hamburger'
import toJson from "enzyme-to-json";
import Adapter from 'enzyme-adapter-react-16';
 
configure({ adapter: new Adapter() });

let props = {
  toggleMenu: jest.fn()
}

// TODO Simulate click of the hamburger icon to toggle open/close
describe('Hamburger Tests', () => {

  it('renders the UI without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render( 
      <Router>
        <Hamburger {...props} />
      </Router>, div);
    ReactDOM.unmountComponentAtNode(div);
  })
  
  it('renders the ui as expected', () => {
    const tree =  mount(
      <Router>
        <Hamburger {...props} />
      </Router>)
    expect(toJson(tree)).toMatchSnapshot();  
  })

})