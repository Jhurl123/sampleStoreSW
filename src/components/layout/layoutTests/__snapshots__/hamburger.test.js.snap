// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hamburger Tests renders the ui as expected 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Hamburger
      toggleMenu={[MockFunction]}
    >
      <button
        className="hamburgerButton"
        onClick={[Function]}
        type="button"
      >
        <div
          className="mainLine "
        />
      </button>
    </Hamburger>
  </Router>
</BrowserRouter>
`;
